hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactingMemStore.java

```diff
@@ -467,7 +498,6 @@ private void doCompact() {
       } catch (Exception e) {
         LOG.debug("Interrupting the MemStore in-memory compaction for store " + getFamilyName());
         Thread.currentThread().interrupt();
-        return;
       } finally {
         releaseResources();
```

Comment:
Make sure the flow of release resources indeed release all resources including closing scanners to avoid memory leak